modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -747,12 +763,180 @@ public void testUpdateGroupWithPUTNotImplementedException(String id, String scim
+
+        Group groupNew = (Group) scimNewGroupObject;
+        Group groupOld = (Group) scimOldGroupObject;
+
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        Mockito.when(userManager.getGroup(anyString(), anyMap())).thenReturn(groupOld);
+        Mockito.when(userManager.patchGroup(anyString(), anyString(), anyMap(), anyMap())).thenReturn(groupNew);
+        SCIMResponse scimResponse = groupResourceManager.updateWithPATCH(existingId, patchRequest,
+                userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_NO_CONTENT);
+    }
+
+    @DataProvider(name = "dataForListWithGET")
+    public Object[][] dataToListWithGET() {
+
+        return new Object[][]{
+                {null, 1, 2, null, null, "PRIMARY", "members", null},
+                {"displayName sw PRIMARY/manager", 1, 2, null, null, "PRIMARY", "members",
+                        null}
+        };
+    }
+
+    @Test(dataProvider = "dataForListWithGET")
+    public void testListWithGET(String filter, Integer startIndexInt, Integer countInt,
+                                String sortBy, String sortOrder, String domainName,
+                                String attributes, String excludeAttributes) {
+
+        SCIMResponse scimResponse = groupResourceManager.listWithGET(userManager, filter, startIndexInt,
+                countInt, sortBy, sortOrder, domainName, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+
+    }
+
+    @Test(dataProvider = "dataForListWithGET")
+    public void testListWithGETProvidedUserManagerHandlerIsNull(String filter, Integer startIndexInt, Integer countInt,
+                                                                String sortBy, String sortOrder, String domainName,
+                                                                String attributes, String excludeAttributes) {
+
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+        SCIMResponse scimResponse = groupResourceManager.listWithGET(null, filter, startIndexInt,
+                countInt, sortBy, sortOrder, domainName, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_INTERNAL_ERROR);
+
+    }
+
+    @DataProvider(name = "dataForListWithPOST")
+    public Object[][] dataToListWithPOST() throws BadRequestException, CharonException, InternalErrorException {
+
+        List<Object> tempList = new ArrayList<>();
+        tempList.add(1);
+        tempList.add(getNewGroup());
+        return new Object[][]{
+
```

Comment:
Unnecessary new line

Reply:
Fixed in commit 34423db0111992d6450b72b5cb1f2b50ab84aec0